lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -3745,15 +3745,38 @@ SPIRVValue *LLVMToSPIRVBase::transIntrinsicInst(IntrinsicInst *II,
     AnnotationDecorations Decorations =
         tryParseAnnotationString(BM, AnnotationString);
 
+    // Translate FPGARegIntel annotations to OpFPGARegINTEL.
```

Comment:
This change seems to be unrelated to the issue being addressed. Can you please comment? Thanks

Reply:
This was something I discovered when running the tests: there are a few cases where this FPGARegIntel annotation is used on struct values, and my change was incorrectly trying to translate those as OpMemberDecorate instead. It seems like in cases where we have an FPGARegIntel annotation we want to translate it to OpFPGARegINTEL regardless of whether we find a struct type, so I thought the cleanest way of doing that would be moving this check up here.

@vmaksimo, from the git log it looks like you originally added this code; is putting it here correct?

Reply:
@dwoodwor-intel Oh, that has been some long time ago, and I missed a big part of changes related to this feature transformation :) 

But I do think that now it's a correct place - it looks logical and has not failed any tests.